lib/grpc/channel.ex

```diff
@@ -10,18 +10,20 @@ defmodule GRPC.Channel do
     * `:host` - server's host to connect
     * `:port` - server's port to connect
     * `:scheme` - scheme of connection, like `http`
+    * `:cred` - credentials used for authentication
```

Comment:
It feels like credentials are separate from the underlying adapter. If a new adapter was needed I think it would use the same credential information chatterbox does. Instead of storing them with adapter specific data I moved them out into the `Channel`.